cmake/px4_add_rust_module.cmake

```diff
@@ -0,0 +1,168 @@
+# are met:
+#
+# 1. Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in
+#    the documentation and/or other materials provided with the
+#    distribution.
+# 3. Neither the name PX4 nor the names of its contributors may be
+#    used to endorse or promote products derived from this software
+#    without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+# OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+# AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+############################################################################
+include(px4_list_make_absolute)
+
+#=============================================================================
+#
+#	px4_add_rust_module
+#
+#	This function builds a static library from a module description.
+#
+#	Usage:
+#		px4_add_module(MODULE <string>
+#			MAIN <string>
+#			[ STACK_MAIN <string> ]
+#			[ STACK_MAX <string> ]
+#			[ COMPILE_FLAGS <list> ]
+#			[ INCLUDES <list> ]
+#			[ DEPENDS <string> ]
+#			[ SRCS <list> ]
+#			[ MODULE_CONFIG <list> ]
+#			[ EXTERNAL ]
+#			[ DYNAMIC ]
+#			)
+#
+#	Input:
+#		MODULE			: unique name of module
+#		MAIN			: entry point
+#		STACK			: deprecated use stack main instead
+#		STACK_MAIN		: size of stack for main function
+#		STACK_MAX		: maximum stack size of any frame
+#		COMPILE_FLAGS		: compile flags
+#		LINK_FLAGS		: link flags
+#		SRCS			: source files
+#		MODULE_CONFIG		: yaml config file(s)
+#		INCLUDES		: include directories
+#		DEPENDS			: targets which this module depends on
+#		EXTERNAL		: flag to indicate that this module is out-of-tree
+#		DYNAMIC			: don't compile into the px4 binary, but build a separate dynamically loadable module (posix)
+#		UNITY_BUILD		: merge all source files and build this module as a single compilation unit
+#
+#	Output:
+#		Static library with name matching MODULE.
+#		(Or a shared library when DYNAMIC is specified.)
+#
+#	Example:
+#		px4_add_module(MODULE test
+#			SRCS
+#				file.cpp
+#			STACK_MAIN 1024
+#			DEPENDS
+#				git_nuttx
+#			)
+#
+function(px4_add_rust_module)
```

Comment:
Is it possible to add just another parameter to "add module"? Like "LANG RuST" or such.... Most of the content looks the same...

Reply:
px4_add_rust_module file removed and instead added LANG parameter into px4_add_module function to define it is RUST module. In case "LANG Rust" is defined (value is case insensitive), the RUST_MOD must be also defined.